human_cancer_analysis/nbconverted/2_identify_generic_genes_pathways.py

```diff
@@ -0,0 +1,778 @@
+print(powers_rank_df.shape)
+powers_rank_df.head()
+
+
+# In[38]:
+
+
+# Count the number of experiments where a given pathway was found to be enriched (qvalue < 0.05)
+total_num_experiments = powers_rank_df.shape[1]
+frac_enriched_pathways = ((powers_rank_df < 0.05).sum(axis=1) / total_num_experiments)
+
+# Rank pathways from 0-50, 50 indicating that the pathways was frequently enriched
+pathway_ranks = frac_enriched_pathways.rank()
+
+powers_rank_stats_df = pd.DataFrame(
+    data={
+        'Fraction enriched': frac_enriched_pathways.values,
+        'Powers Rank':pathway_ranks.values
+    },
+    index=powers_rank_df.index
+)
+powers_rank_stats_df.sort_values(by="Powers Rank", ascending=False).head()
+
+
+# In[39]:
+
+
+# Save reference file for input into comparison
+powers_rank_processed_filename = os.path.join(
+    base_dir,
+    dataset_name,
+    "data",
+    "metadata",
+    "Hallmarks_qvalues_GSEAPreranked_processed.tsv"
+)
+
+powers_rank_stats_df.to_csv(powers_rank_processed_filename, sep="\t", )
+
+
+# In[53]:
+
+
+figure_filename = f"pathway_ranking_{col_to_rank_pathways}.svg"
+
+process.compare_pathway_ranking(
+    summary_pathway_ranks,
+    powers_rank_processed_filename,
+    figure_filename
+)
+
+
+# * Our method ranked pathways using median adjusted p-value score across simulated experiments.
+# * Powers et. al. ranked pathways based on the fraction of experiments they had adjusted p-value < 0.05.
+# * Next let's try to use the fraction of adjusted p-value < 0.05 for our method and re-compare (below)
+# 
+# **Takeaway:**
+# * Previously we compared pathway ranks obtained from (recount2)-trained VAE model vs pathway ranking based on manual curation using Powers et. al. This [PR](https://github.com/ajlee21/generic-expression-patterns/blob/807377d76f63b6282c62255d7b160feb8585e0e2/human_analysis/2_identify_generic_genes_pathways.ipynb) shows that there was no correlation.
+# 
+# * Here we are comparing pathway ranks obtained from a (Powers et. al.)-trained VAE model vs pathway ranking based on manual curation using Powers et. al datasets. Based on this correlation plot there is a high correlation.
+
+# **Conclusion:**
+# 
+# * We can get relatively similar generic genes training our VAE model on a cancer-specific dataset (Powers et. al.) and testing on a genenal dataset (Crow et. al.). These generic genes are not *that* context-specific at the extremes
+# 
+# * We get very different generic pathways training our VAE model on a general dataset (recount2) and testing on a cancer-specific dataset (Powers et. al.) -- see [this analysis](../human_analysis/2_identify_generic_genes_pathways.ipynb). But we get very similar generic pathways training our VAE model on a cancer-specific dataset (Powers et. al.) and testing on a cancer-specific dataset (Powers et. al.). This indicates that generic pathways are more context specific.
+# 
+# * I need to think about about why there is a difference in genes vs pathways.
```

Comment:
Maybe the strong signal in pathways is driven by small expression changes in lots of downstream genes, so when you look on an individual gene level the results don't really reflect the change in context, but when you look at pathways as a whole you're better able to differentiate between contexts?

I think maybe this also makes sense if you think about it in the context of the paper Ben and I presented at journal club (https://www.biorxiv.org/content/10.1101/2020.07.30.228296v1): different experimental conditions lead to changes in the activity of signaling pathways, which lead to changes in the expression of the genes these signaling pathways regulate as a group. Since the genes are regulated as a group, you'd see (possibly small) coordinated changes in expression that aren't necessarily significant on the individual gene level.

Anyway, I'm just throwing some ideas out there (and I really like the idea of thinking about this on a pathway level, especially in the context of cancer - although I know you're firmly on team microbe and probably aren't too interested in cancer applications 😂 ). I think you'd need to do further analyses to really confirm (or deny) any of what I'm saying, but maybe something in here will help you form a mental model of how this is working, at least.

Reply:
This is great insight! Thank you for thinking about this. I'm so glad that you find the results exciting! I definitely need to think more about this once the repo is re-organized. But I was also thinking about your paper after I got these results:) 

ps. I shall always be team microbe...especially given I started in viral genomics before grad school. But I'm honestly team everything:)